### ISSUE TITLE
#[must_use] on methods is experimental

### ISSUE THREAD

**Issue by dtolnay** [maciejhirsz/json-rust#119]
I see this warning compiling json 0.11.9 with rustc 1.21.0-nightly (003a929f9 2017-08-26):

```
warning: `#[must_use]` on methods is experimental (see issue #43302)
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/json-0.11.9/src/value/mod.rs:370:5
    |
370 | /     pub fn push<T>(&mut self, value: T) -> Result<()>
371 | |     where T: Into<JsonValue> {
372 | |         match *self {
373 | |             JsonValue::Array(ref mut vec) => {
...   |
378 | |         }
379 | |     }
    | |_____^
    |
    = help: add #![feature(fn_must_use)] to the crate attributes to enable
```
